### PR TITLE
Dev mode

### PR DESCRIPTION
**Pull request by andorp**
Fixes: #845 
Fixes: #846

### CODE REVIEW

File: src/Frontend/Core.hs [hunk 1]
```diff
@@ -642,8 +642,8 @@ cancelButton p = a_ [class_ "btn", href_ $ redirectOf p ()] "Abbrechen"
 -- the entire sequence of requests the client presents the same cookie, not only over those
 -- sub-sequences where the user is logged-in.)
 data Frame body
-    = Frame { _frameUser :: User, _frameBody :: body, _frameMessages :: [StatusMessage] }
-    | PublicFrame               { _frameBody :: body, _frameMessages :: [StatusMessage] }
+    = Frame { _frameUser :: User, _frameBody :: body, _frameMessages :: [StatusMessage], _frameDevMode :: Bool }
+    | PublicFrame               { _frameBody :: body, _frameMessages :: [StatusMessage], _frameDevMode :: Bool }
```
Comment (fisx) [inline src/Frontend/Core.hs]: It's looking more and more like the `_frameUser` should be a `Maybe` and there should be only one constructor, but let's not try that change here.

File: src/Frontend/Core.hs [hunk 2]
```diff
@@ -850,20 +850,21 @@ renderStatusMessage :: (Monad m) => StatusMessage -> HtmlT m ()
...
             span_ [class_ "main-footer-blurb"] $ do
                 "Made with \x2665 by Liqd"
-                replicateM_ 5 $ toHtmlRaw nbsp
-                toHtml Config.releaseVersion
-                replicateM_ 5 $ toHtmlRaw nbsp
-                a_ [Lucid.onclick_ "createPageSample()"]
-                    "[create page sample]"  -- see 'Frontend.createPageSamples" for an explanation.
+                when devmode $ do
+                    replicateM_ 5 $ toHtmlRaw nbsp
+                    toHtml Config.releaseVersion
```
Comment (fisx) [inline src/Frontend/Core.hs]: i think the release should be in the production page as well.

Comment (fisx) [inline src/Frontend/Core.hs]: i just saw that the ticket says otherwise.  i disagree, but ok, never mind.

File: src/Frontend/Core.hs (hunk 1 above)
Comment (andorp) [inline src/Frontend/Core.hs]: I would say we could introduce a `FrameKind` type which is isomorph to the Maybe type.

**Pull request merged**